### PR TITLE
Enhancement - Add tags to Jira tickets for finding groups

### PR DESCRIPTION
**Pull request by schuman0**
## Please Note: We consider the open-source version of DefectDojo to be feature complete. We're open to your suggestions and feedback, but we do not plan to add or accept new features in the future for supportability concerns.

**Description**

Adding support for pushing tags to Jira in finding groups.

**Test results**

Jira tickets for finding groups now include tags correctly.

**Documentation**

N/A

**Checklist**

This checklist is for your information.

- [ ] Make sure to rebase your PR against the very latest `dev`.
- [ ] Features/Changes should be submitted against the `dev`.
- [ ] Bugfixes should be submitted against the `bugfix` branch.
- [ ] Give a meaningful name to your PR, as it may end up being used in the release notes.
- [ ] Your code is flake8 compliant.
- [ ] Your code is python 3.11 compliant.
- [ ] If this is a new feature and not a bug fix, you've included the proper documentation in the docs at https://github.com/DefectDojo/django-DefectDojo/tree/dev/docs as part of this PR.
- [ ] Model changes must include the necessary migrations in the dojo/db_migrations folder.
- [ ] Add applicable tests to the unit tests.
- [ ] Add the proper label to categorize your PR.

### CODE REVIEW

File: dojo/jira_link/helper.py [hunk 1]
```diff
@@ -535,6 +535,12 @@ def get_tags(obj):
         if obj_tags:
             for tag in obj_tags:
                 tags.append(str(tag.name))
+    if type(obj) == Finding_Group:
+        for finding in obj.findings.all():
+            obj_tags = finding.tags.all()
+            if obj_tags:
+                for tag in obj_tags:
+                    tags.append(str(tag.name))
```
Comment (Maffooch) [inline dojo/jira_link/helper.py:543]: Can this be updated to remove potential duplicates in the list

**Review by Maffooch — CHANGES_REQUESTED** [review]

File: dojo/jira_link/helper.py (hunk 1 above)
Comment (Maffooch) [reply dojo/jira_link/helper.py:544]: @schuman0 I was talking about the finding groups. Findings and engagements already cannot have the same tag. If a Finding A and Finding B have the same tag, this code will put two of the same tag in the same list and attempt to push to jira.

Comment (schuman0) [reply dojo/jira_link/helper.py]: My apologies. I added it in the wrong location

**Review by Maffooch — APPROVED** [review]

**Review by mtesauro — APPROVED** [review]
Approved

**Review by devGregA — APPROVED** [review]

**Review by grendel513 — APPROVED** [review]

**Pull request merged**